### PR TITLE
feat(modal): add component support

### PR DESCRIPTION
**Pull request by wesleycho**
- Add support for `component` option

This should implement #5683.

### CODE REVIEW

File: src/modal/docs/readme.md [hunk 1]
```diff
@@ -35,15 +35,29 @@ The `$uibModal` service has only one method: `open(options)`.
   _(Type: `boolean`, Default: `false`)_ -
   When used with `controllerAs` & set to `true`, it will bind the $scope properties onto the controller.
 
+* `component`
+  _(Type: `string`, Example: `myComponent`)_ -
+  A string reference to the component to be rendered that is registered with Angular's compiler. If using a directive, the directive must have `restrict: 'E'` and a template or templateUrl set.
```
Comment (icfantv) [inline src/modal/docs/readme.md]: I think this implies that this is a BC, no?

Comment (wesleycho) [inline src/modal/docs/readme.md]: This is a completely new option - it isn't supported currently, so it isn't a BC at all.

File: src/modal/test/modal.spec.js [hunk 2]
```diff
@@ -131,6 +131,16 @@ describe('$uibModal', function() {
           elem.focus();
         }
       };
+    }).component('fooBar', {
+      bindings: {
+        resolve: '<',
```
Comment (icfantv) [inline src/modal/test/modal.spec.js]: do we need to mention in the docs that this and the modal instance are one-way bound?  realizing that it doesn't apply to object contents...

Comment (icfantv) [inline src/modal/test/modal.spec.js]: gah, nm.  i just realized this was a test...

**Comment by icfantv** [comment]
LGTM

**Pull request closed without merging**